extensions/container-image/container-image-s2i/deployment/src/main/java/io/quarkus/container/image/s2i/deployment/PackageUtil.java

```diff
@@ -0,0 +1,216 @@
+
+package io.quarkus.container.image.s2i.deployment;
+
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.jsoup.helper.StringUtil;
+
+import io.dekorate.DekorateException;
+import io.dekorate.deps.commons.compress.archivers.tar.TarArchiveEntry;
+import io.dekorate.deps.commons.compress.archivers.tar.TarArchiveOutputStream;
+
+public class PackageUtil {
```

Comment:
Can we get some comments in this?

Mentioning it because this looks like where most of the problems could arise - the rest of the changes seem pretty straighforward, but this one is a mouth full :)

Reply:
This is copied from dekorate and modified so that it can support custom entry prefix paths.
This should be fixed upstream. See: https://github.com/dekorateio/dekorate/issues/621